tests/director/tests/application/application_templates_api_test.go

```diff
@@ -102,13 +106,187 @@ func TestCreateApplicationTemplate(t *testing.T) {
 		t.Log("Create application template")
 		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, createApplicationTemplateRequest, &output)
 		defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenantID, output)
+		require.NoError(t, err)
 
 		// THEN
 		require.Equal(t, output.Labels[conf.ApplicationTemplateProductLabel], productLabelValue)
-		require.NoError(t, err)
 		require.NotEmpty(t, output.ID)
 	})
 
+	t.Run("Success for regional template with product label created with certificate", func(t *testing.T) {
+		// GIVEN
+		ctx := context.Background()
+
+		productLabelValue := []interface{}{"productLabelValue"}
+		appTemplateName1 := fixtures.CreateAppTemplateName("app-template-name-product-1")
+		appTemplateInput1 := fixtures.FixRegionalApplicationTemplate(appTemplateName1, region1, regionPlaceholderJSONPath1)
+		appTemplateInput1.Labels[conf.ApplicationTemplateProductLabel] = productLabelValue
+		appTemplate1, err := testctx.Tc.Graphqlizer.ApplicationTemplateInputToGQL(appTemplateInput1)
+		require.NoError(t, err)
+
+		createApplicationTemplateRequest1 := fixtures.FixCreateApplicationTemplateRequest(appTemplate1)
+		output1 := graphql.ApplicationTemplate{}
+
+		appTemplateName2 := fixtures.CreateAppTemplateName("app-template-name-product-2")
+		appTemplateInput2 := fixtures.FixRegionalApplicationTemplate(appTemplateName2, region2, regionPlaceholderJSONPath1)
+		appTemplateInput2.Labels[conf.ApplicationTemplateProductLabel] = productLabelValue
+		appTemplate2, err := testctx.Tc.Graphqlizer.ApplicationTemplateInputToGQL(appTemplateInput2)
+		require.NoError(t, err)
+
+		createApplicationTemplateRequest2 := fixtures.FixCreateApplicationTemplateRequest(appTemplate2)
+		output2 := graphql.ApplicationTemplate{}
+
+		// WHEN
+		t.Log("Create first regional application template")
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, createApplicationTemplateRequest1, &output1)
+		defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenantID, output1)
+		require.NoError(t, err)
+
+		t.Log("Create second regional application template")
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, createApplicationTemplateRequest2, &output2)
+		defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenantID, output2)
+		require.NoError(t, err)
+
+		// THEN
+		require.Equal(t, output1.Labels[conf.ApplicationTemplateProductLabel], productLabelValue)
```

Comment:
This is a global comment about all occurrences of require.Equal():
Swap places of expected and actual values in the Equal function. Currently as they are set, the troubleshooting when a test fails will be more complicated.